spec/beta/rate_spec.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe EasyPost::Beta::Rate do
+  describe '.retrieve_stateless_rate' do
+    it 'retrieve all stateless rates' do
+      stateless_rates = described_class.retrieve_stateless_rates(Fixture.basic_shipment)
+
+      expect(stateless_rates).to all(be_an_instance_of(EasyPost::Rate))
+    end
+
+    describe '.get_lowest_stateless_rate' do
+      it 'get the lowest stateless rate' do
```

Comment:
In a perfect world, this would be broken up into three separate `it` blocks so that you could have the it blocks say something like (I don't think our other tests are setup this way, so this is fine for now):

1. gets the lowest stateless rate without filters
2. returns an error when an invalid carrier is used
3. returns an error when an invalid service is used

The it block should define what the test is doing (the branch) that's being tested of the greater `describe` block which names a function/method. This way it can all be grouped together.
```suggestion
      it 'gets the lowest stateless rate for various combinations of filters' do
```

Reply:
We could do that, but it would generate three cassettes that have the same content.